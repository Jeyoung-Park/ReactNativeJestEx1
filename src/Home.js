import React, {useState} from 'react';
import {SafeAreaView, Text, TextInput, View} from 'react-native';

// const Home = () => {
//   const [data, setData] = useState('test');

//   const change = x => {
//     // return x * 10;
//     setData(x * 10);
//   };

//   return (
//     <SafeAreaView
//       style={{flex: 1, justifyContent: 'center', alignItems: 'center'}}>
//       <Text style={{fontSize: 20}}>Home</Text>
//     </SafeAreaView>
//   );
// };

class Home extends React.Component {
  constructor() {
    super();
    this.state = {
      data: 'test',
    };
  }

  change(x) {
    // return x * 10;
    this.setState({data: x * 10});
  }

  render() {
    return (
      <SafeAreaView
        style={{flex: 1, justifyContent: 'center', alignItems: 'center'}}>
        {/* <Text style={{fontSize: 20}}>Home</Text> */}
        <TextInput
          testID={'username'}
          onChangeText={text => this.change(text)}
          placeholder="Enter Username"
          style={{backgroundColor: 'gray', marginBottom: 15}}
        />
        <TextInput
          testID={'password'}
          onChangeText={text => this.change(text)}
          placeholder="Enter Password"
          style={{backgroundColor: 'gray', marginBottom: 15}}
        />
      </SafeAreaView>
    );
  }
}

export default Home;
