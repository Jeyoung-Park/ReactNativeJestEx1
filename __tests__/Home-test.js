import 'react-native';
import React from 'react';
import Home from '../src/Home';
import renderer from 'react-test-renderer';
import {expect} from '@jest/globals';

test('Home snapShot', () => {
  const snap = renderer.create(<Home />).toJSON();

  expect(snap).toMatchSnapshot();
});
