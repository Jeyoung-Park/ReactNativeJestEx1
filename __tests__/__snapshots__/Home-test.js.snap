// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapShot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Home
  </Text>
</RCTSafeAreaView>
`;
